### PR TITLE
PLAT-277 GitHub Action to Deploy SS to S3

### PR DESCRIPTION
**Pull request by james-nduka**
## Changes

- Add workflow job in order to deploy stylesheets to  `s3://imgarena-static/` bucket based on merge to Master
- A step included to Invalidate the previous stylesheets at the CloudFront Edge Cache
- Do we need to add the tests here or is that covered by pr #2 ?

## Reference

https://imgarenahub.atlassian.net/browse/PLAT-277?atlOrigin=eyJpIjoiZjE4MjM3NmE2MzA4NDRmYmE4MjdiNDgxMDQwODY5YTQiLCJwIjoiaiJ9

### CODE REVIEW

**Review by footerz — APPROVED** [review]

File: .github/workflows/deployment-master.yml [hunk 1]
```diff
@@ -1,46 +1,40 @@
...
+          export INV_STATUS=`aws cloudfront get-invalidation --id $INV_ID --distribution-id $PROD_STATIC_DIST |  jq -r '.Invalidation.Status'`
+          if [ "$INV_STATUS" == "Completed" ]; then
+            echo "Invalidation of Stylesheets Cache. Status=$INV_STATUS"
+          else
+            echo "ERROR: Invalidation of Cache has not finished successfully. STATUS=$INV_STATUS"
+            exit 1
+          fi
+      - name: Deploy Stylesheets
+        run: |
+          aws s3 sync stylesheets s3://imgarena-static/stylesheets --acl private --delete --cache-control max-age=86400
```
Comment (akosveres) [inline .github/workflows/deployment-master.yml:40]: I think the cache-control needs an additional value based on what we use in the event centers
```suggestion
          aws s3 sync stylesheets s3://imgarena-static/stylesheets --acl private --delete --cache-control "public, max-age=86400"
```

Event centre link to the github actions https://github.com/IMGARENA/ufc-event-centre/blob/master/.github/workflows/deployment-master.yml#L41

File: .github/workflows/deployment-master.yml [hunk 2]
```diff
@@ -1,46 +1,40 @@
...
-#           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-#           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-#           aws-region: eu-west-1
-#           role-to-assume: 'arn:aws:iam::460928140709:role/frs-dev-cicd'
-#           role-duration-seconds: 1200
-
-#       - name: Deploy to AWS
-#         run: |
-#           aws s3 sync dist s3://imgarena-apps-static-assets-frs-dev/${SPORT}/latest --acl private --delete
+      - name: Invalidate Cache Stylesheets
```
Comment (akosveres) [inline .github/workflows/deployment-master.yml:27]: I believe the cache should be invalidated after we push the new files, to make sure calls done after invalidating and before new files are pushed are not using the old stylesheets. Of course local cache of files will still happen, but we discussed during the initial call that it's okay, we'll deal with that later.

(Leaving this comment so our discussion during stand-up does not get lost)

File: .github/workflows/deployment-master.yml (hunk 1 above)
Comment (akosveres) [inline .github/workflows/deployment-master.yml]: Based on the requirements in the ticket, the directory in s3 should be `event-centre-stylesheets` (seems the details and AC have different URLs, which is confusing, but we agreed the end URL should be https://static.imgarena.com/event-centre-stylesheets/*.css)

File: .github/workflows/deployment-master.yml (hunk 2 above)
Comment (james-nduka) [reply .github/workflows/deployment-master.yml:31]: Resolved.

File: .github/workflows/deployment-master.yml (hunk 1 above)
Comment (james-nduka) [reply .github/workflows/deployment-master.yml]: Updated

**Review by akosveres — APPROVED** [review]

**Pull request merged**